[Docs] Add LocalNet issues and solutions to FAQ

h5law opened the pull request:
## Description

Fix errors with the `Makefile` by removing the `--build` flag from the incorrect position in the `client_start` command, and add the `rebuild_client_start` command as well.

Add a description of an issue relating to starting the LocalNet with SELinux and its solution to `docs/development/FAQ.md`. This will be useful when `make compose_and_watch` throw errors, these have also been included in the file..

## Issue

No issue documentation update

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [x] Documentation
- [ ] Other <!-- add details here if it a different type of change -->

## List of changes

- Add issue and solution to FAQ for SELinux blocking the docker container from accessing local files to start the LocalNet
- Remove the `--build` flag from the `client_start` command in the `Makefile` and add the `rebuild_client_start` command with the `--build` flag in the correct position

## Testing

- [x] `make develop_test`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my changes using the available tooling
- [x] I have updated the corresponding CHANGELOG

### If Applicable Checklist
- [x] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

Olshansk commented:
@h5law Make sure to update these in the future:

<img width="358" alt="Screenshot 2022-12-23 at 1 52 02 PM" src="https://user-images.githubusercontent.com/1892194/209394090-d5ceffea-8acd-4416-afd7-bee4c3204028.png">

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+node3.consensus exited with code 2
+node4.consensus exited with code 2
+```
+* **Solution**: A temporary fix would be to run
````
```suggestion

* **Solution**: A temporary fix would be to run:

```

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+```
+$ su -c "setenforce 0"
+```
+Whereas a permenant approach would be to allow the docker container access to the local repository
````
```suggestion

Whereas a permenant approach would be to allow the docker container access to the local repository:

```

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+```
+* **Solution**: A temporary fix would be to run
+```
+$ su -c "setenforce 0"
````
No need to add `$`. 

I used to do this myself in the past as well, but GitHub let's you copy-pasta, do the bash commands don't help.

Ditto everywhere below.

h5law commented:
> @h5law Make sure to update these in the future:

Yeah completely passed my mind sorry. I'll make sure to do it all next time.

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+node4.consensus exited with code 2
+```
+* **Solution**: A temporary fix would be to run
+```
````
Replace this:
 \```
with this:
\```bash

This applies the proper syntax highlighting. Ditto everywhere below.

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+```
+Whereas a permenant approach would be to allow the docker container access to the local repository
+```
+$ cd pocket
+$ sudo chcon -Rt svirt_sandbox_file_t ./
````
```suggestion
sudo chcon -Rt svirt_sandbox_file_t ./pocket
```

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+$ sudo chcon -Rt svirt_sandbox_file_t ./
+```
+
+See [this stackoverflow post](https://stackoverflow.com/questions/24288616/permission-denied-on-accessing-host-directory-in-docker) for more details
````
```suggestion
See [this stackoverflow post](https://stackoverflow.com/questions/24288616/permission-denied-on-accessing-host-directory-in-docker) for more details.
```

Olshansk commented on docs/development/FAQ.md:
```diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+
+## Unable to create a LocalNet client
+
+* **Issue**: When running `make client_start` you get the error message
```
```suggestion
* **Issue**: When running `make client_start` you get the error message:
```

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+    docker-compose -f build/deployments/docker-compose.yaml up -d client
+```
+
+_NOTE: Consider turning of the `gofmt` in your IDE to prevent unexpected formatting_
````
Does this imply that we should run a one-time `gofmt` across the whole codebase?

h5law replied:
I didn't add this comment it was there before

Olshansk commented on docs/development/FAQ.md:
````diff
@@ -4,7 +4,59 @@ A list of common issues & resolutions shared by the V1 contributors
+    docker-compose -f build/deployments/docker-compose.yaml up -d client --build
+```
+
+Remove the `--build` so the lines look like:
````
Can you explain why this happens? The Makefile should just work for everyone out of the box, so there must be a better solution.

h5law replied:
@Olshansk I'm not too sure why it happens - I haven't got much experience with docker so I couldn't really explain the reasoning but I found the error would only occur with the `--build` flag present. Without the `--build` flag the build still occurs _I assume_ when there is no current client build already.

h5law replied:
@Olshansk I have found the reason, according to this documentation https://docs.docker.com/compose/profiles/

I believe the line in the `Makefile` should be 
`docker-compose -f build/deployments/docker-compose.yaml up --build -d client` 
where the `build` profile comes before the `client` service

I can confirm that this works, _on my machine_ so I am not willing to submit a PR to change the Makefile if it is working for everyone else but I will update the FAQ with this reasoning.

Olshansk replied:
This just seems to be incorrect.

See a couple of our other targets for reference

```makefile
.PHONY: compose_and_watch
compose_and_watch: docker_check db_start monitoring_start ## Run a localnet composed of 4 consensus validators w/ hot reload & debugging
	docker-compose -f build/deployments/docker-compose.yaml up --force-recreate node1.consensus node2.consensus node3.consensus node4.consensus

.PHONY: rebuild_and_compose_and_watch
rebuild_and_compose_and_watch: docker_check db_start monitoring_start ## Rebuilds the container from scratch and launches compose_and_watch
	docker-compose -f build/deployments/docker-compose.yaml up --build --force-recreate node1.consensus node2.consensus node3.consensus node4.consensus
```

How about we?
1. Remove `--build` from `client_start`
2. Add `rebuild_client_start` with the flag in the appropriate location
3. Remove this FAQ (i.e. keep it simple)

Olshansk reviewed (changes requested):
@h5law Make sure to re-request a review in the future to make sure the reviewer knows the PR is ready for a second round:

![Screenshot 2022-12-27 at 11 43 18 AM](https://user-images.githubusercontent.com/1892194/209697098-3d51fa58-fb4f-423b-bc26-f958b38e1bdf.png)

Also, if there's a suggestion or comment that you accepted (via Github's commit suggestion or not), feel free to resolve it yourself. We have implicit trust on the team that if a comment was "trivial", the author tended to it appropriately :)

![Screenshot 2022-12-27 at 11 45 24 AM](https://user-images.githubusercontent.com/1892194/209697143-76ab56e9-c46a-4e67-9d3d-7f6265599099.png)

h5law commented:
@Olshansk Removed the FAQ question and updated the `Makefile` (removed `--build` and added the `rebuild_client_start` command). Should be all good to go 😄

Olshansk commented:
LGTM!

@h5law Do you have permissions to `squash & mege` yourself? If so, go ahead and do so. Our team's pattern is to copy-paste the PR description into the squashed commit message.

h5law commented:
> Do you have permissions to `squash & mege` yourself?

@Olshansk No, I don't have permission to merge the PR myself. But I'll remember the pattern for when I can 😄
